### PR TITLE
Fix for reporting correct bundle size 

### PR DESCRIPTION
**Pull request by sonalivdeshpande**
The bundle size comparison tool is currently comparing baseline against main branch. The baseline for pull requests opened against the next branch are incorrect. This pull request aims to fix the baseline comparison by setting an env variable `TARGET_BRANCH_NAME` as branch that is the target of a pull request. By default, the env variable is set to `main`.

[AB#1570](https://dev.azure.com/fluidframework/internal/_workitems/edit/1570)

### CODE REVIEW

File: tools/pipelines/templates/build-npm-package.yml [hunk 1]
```diff
@@ -148,6 +148,7 @@ stages:
                 publish=${{ variables.publish }}
                 canRelease=${{ variables.canRelease }}
                 publishNonScopedPackages=${{ variables.publishNonScopedPackages }}
+                targetbranchName = $(System.PullRequest.TargetBranch)
```
Comment (ChumpChief) [inline tools/pipelines/templates/build-npm-package.yml:151]: Nit:
```suggestion
                targetBranchName = $(System.PullRequest.TargetBranch)
```

**Review by ChumpChief — COMMENTED** [review]
I think it looks reasonable - will be interested to see it run against this PR :)

**Comment by sonalivdeshpande** [comment]
> I think it looks reasonable - will be interested to see it run against this PR :)

bundle size not reported :(

**Comment by ChumpChief** [comment]
> > I think it looks reasonable - will be interested to see it run against this PR :)
> 
> bundle size not reported :(

Might be due to that other bug we were observing, that it only successfully reports bundle size if the first run succeeds.  Maybe try opening a fresh PR with this change and see if you can get a clean run?

**Comment by curtisman** [comment]
Can we write some description for the PR?  Not all people have access to azure  work item

**Comment by curtisman** [comment]
> > > I think it looks reasonable - will be interested to see it run against this PR :)
> > 
> > 
> > bundle size not reported :(
> 
> Might be due to that other bug we were observing, that it only successfully reports bundle size if the first run succeeds. Maybe try opening a fresh PR with this change and see if you can get a clean run?

I think there is some heuristic to not report to GH if there is no difference

**Comment by sonalivdeshpande** [comment]
For this PR, there was no difference in bundle size -- `No size changes detected, skipping posting PR comment`. 

For one of the PR against which the bundle size was not reported: https://github.com/microsoft/FluidFramework/pull/11491 -- the logs are as follows: 

`2022-08-10T19:01:56.0187770Z The baseline commit for this PR is b2c560ab996ad4d3f5e4c84e35320502045e2aec`
`2022-08-10T19:01:56.0188737Z Trying backup baseline commit e87851120f912340ee8e63987560c3b78f89f52b`
`2022-08-10T19:01:56.0189767Z Trying backup baseline commit ad7b27a50d04b53ac1b92db95f1a6e1eff9fbe3e`
`2022-08-10T19:01:56.0190326Z Found baseline build with id: 84555`
 `No size changes detected, skipping posting PR comment`

Backup baseline commit is looked up when baseline build is undefined or successful baseline build does not have the needed build artifacts.

File: tools/bundle-size-tools/src/utilities/gitCommands.ts [hunk 2]
```diff
@@ -9,9 +9,9 @@ import { execSync } from 'child_process';
  * Gets the commit in main that the current branch is based on.
```
Comment (ChumpChief) [inline tools/bundle-size-tools/src/utilities/gitCommands.ts:9]: Comment needs updating
```suggestion
 * Gets the commit in the target branch that the current branch is based on.
```

**Review by ChumpChief — APPROVED** [review]
One nit on a comment, but looks good!

**Pull request merged**